src-tools/netutils/asio_send_block/asio_send_block.cpp

```diff
@@ -46,40 +153,140 @@ class c_maintask {
 		<<std::endl;
 }
 
-int c_maintask::run(int argc, char *argv[])
-{
-	if (argc < 3) {
+void c_maintask::print_usage_rpc_cmd() const {
+	std::cout << "secret1 SEND [args for normal run]  <--- this text, ending with a NEW LINE.\n";
+	std::cout << "secret1 SEND 127.0.0.1 9000  999000 foo 1472  -1  \n";
+	std::cout << "secret1 SEND 127.0.0.1 9000  999000 foo 8972  -1 5 -1 (no time limit, 5 seconds, no GB limit) \n";
+}
+
+void c_maintask::print_usage() const {
+		std::cout << std::endl;
 		std::cout << "Usage: ./client <host> <port> <max_speed> SendCommand" << std::endl;
-		std::cout << "e.g.:  ./client <host> <port> <max_speed> <msg> <msgbytes> <count> " << std::endl;
+		std::cout << "e.g.:  ./client <host> <port> <max_speed> <msg> <msgbytes> <count> [maxTime sec] [maxData GB]" << std::endl;
 		std::cout << "e.g.:  ./client 127.0.0.1 9000  999000    foo   1500       -1 " << std::endl;
+		std::cout << "e.g.:  ./client 127.0.0.1 9000  999000    foo   1500       -1 10 -1 " << std::endl;
+		std::cout << "\nor start server accepting remote RPC commands: \n";
+		std::cout << "e.g.:  ./client remote_cmd 192.168.70.17 9000 " << std::endl;
+		std::cout << "or instead allow remote access (remote controll of this sending, WARNING can make your computer spam/DDoS other computer) " << std::endl;
+		std::cout << "e.g.:  ./client remote <listen_on_ip><port>  <authorized_ip> <max_time_hours> <pass>  " << std::endl;
+		std::cout << "e.g.:  ./client remote 192.168.70.17 9011   192.168.70.16   72 secret1" << std::endl;
+		std::cout << "e.g.:  ./client remote 192.168.70.17 9011   0.0.0.0         4  secret1" << std::endl;
+		std::cout << "e.g.:  ./client remote       0.0.0.0 9011   0.0.0.0         4  secret1" << std::endl;
+		std::cout << "then send TCP text like e.g.: \n";
+		print_usage_rpc_cmd();
+		std::cout << std::endl;
+}
+
+std::string c_maintask::run_rpc_command_string(const std::string & rpc_cmd) {
+	std::string ret;
+	ret += "RPC-result ";
+	std::cout << "RPC command to execute: [" << rpc_cmd << "]\n" << std::endl;
+	std::vector<std::string> args = explode( remove_extra_whitespace(rpc_cmd) , ' ');
+	std::cout << "RPC command args size: " << args.size() << std::endl;
+
+	try {
+		std::string password_given = args.at(0);
+		std::string word_SEND = args.at(1);
+
+		if (m_rpc_password.size()<3) throw std::runtime_error("insecure (short) or missing RPC password (configured here)");
+		if (password_given != m_rpc_password) throw std::runtime_error("invalid RPC password given");
+		if (word_SEND != "SEND") throw std::runtime_error("word 'SEND'");
+
+		args.erase( args.begin() , args.begin()+2 );
+
+	} catch(const std::exception &ex) {
+		_warn("Can not parse RPC: " << ex.what() << " rpc was [" << rpc_cmd << "]");
+		throw ;
+	}
+
+	args.insert( args.begin(), "program_name" );
+
+	size_t argc = args.size();
+	const char ** argv = new const char * [ argc ];
```

Comment:
use unique_ptr

Reply:
hmm dunno, better vector of raw ptr then.
thought it's not that needed in low level

using  std::vector<const char*> argv( argc , nullptr );